org.eclipse.jdt.core/search/org/eclipse/jdt/core/search/SearchPattern.java

```diff
@@ -2453,10 +2453,19 @@ private static char[][] enclosingTypeNames(IType type) {
 		case IJavaElement.FIELD:
 		case IJavaElement.INITIALIZER:
 		case IJavaElement.METHOD:
+			char[] typeName = IIndexConstants.ONE_STAR;
+			try {
+				String superclassName = type.getSuperclassName();
```

Comment:
Since we have an anonymous class, `getSuperclassName()` yields the interface or class that is being implemented resp. extended. No need to check interfaces (which I initially thought we also had to do).